src/test/compile-fail/associated-const-in-trait.rs

```diff
@@ -0,0 +1,24 @@
+// Copyright 2017 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// #29924
+
+#![feature(const_fn, associated_consts)]
+
+trait Trait {
+    const N: usize;
+}
+
+impl Trait {
+    //~^ ERROR E0038
```

Comment:
Could you use actual error messages instead of codes outside of `E0___.rs` tests?

Reply:
Thank you for your review. Updated and rebased.

Reply:
@topecongiro could you do that change in all `compile-fail` files?

Reply:
@estebank Update tests and reabased. Also fixed the issue number (#39953 -> #39553).